RDKit fails to parse "M RAD" lines were radical is 0

According to the Spec, 0 is a valid value (no radical), but RDKit only supports 1,2,3, and fails parsing when a 0 is found:

![image](https://github.com/rdkit/rdkit/assets/7498185/2e76a933-ce9f-479e-9e32-5c0e9c097111)


```python3
In [1]: mb = """
   ...:      RDKit          2D
   ...: 
   ...:   1  0  0  0  0  0  0  0  0  0999 V2000
   ...:     0.0000    0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   ...: M  RAD  1   1   0
   ...: M  END
   ...: """

In [2]: m = Chem.MolFromMolBlock(mb)
[13:57:28] Unrecognized radical value 0 for atom 0 on line 6


In [3]: assert m is not None
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
Cell In[3], line 1
----> 1 assert m is not None

AssertionError: 

```

--- a/Code/GraphMol/FileParsers/MolFileParser.cpp
+++ b/Code/GraphMol/FileParsers/MolFileParser.cpp
@@ -422,6 +422,10 @@ void ParseRadicalLine(RWMol *mol, const std::string &text, bool firstCall,
       spos += 4;
 
       switch (rad) {
+        case 0:
+          // This shouldn't be required, but let's make sure.
+          mol->getAtomWithIdx(aid - 1)->setNumRadicalElectrons(0);
+          break;
         case 1:
           mol->getAtomWithIdx(aid - 1)->setNumRadicalElectrons(2);
           break;

--- a/Code/GraphMol/FileParsers/file_parsers_catch.cpp
+++ b/Code/GraphMol/FileParsers/file_parsers_catch.cpp
@@ -7025,3 +7025,22 @@ TEST_CASE("FragmentSgroupTest", "[bug][reader]") {
     }
   };
 }
+
+TEST_CASE(
+    "GitHub Issue #7256: RDKit fails to parse \"M RAD\" lines with were radical is 0",
+    "[bug]") {
+  const auto mb = R"CTAB(
+     RDKit          2D
+
+  1  0  0  0  0  0  0  0  0  0999 V2000
+    0.0000    0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
+M  RAD  1   1   0
+M  END
+)CTAB";
+
+  std::unique_ptr<RWMol> mol(MolBlockToMol(mb));
+  REQUIRE(mol);
+  REQUIRE(mol->getNumAtoms() == 1);
+  const auto at = mol->getAtomWithIdx(0);
+  CHECK(at->getNumRadicalElectrons() == 0);
+}

